### PR TITLE
refactor(trade): move hooks to updaters

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Context: https://cowservices.slack.com/archives/C0361CDG8GP/p1689162210274279?thread_ts=1689160416.268759&cid=C0361CDG8GP

What is changed:
1. Moved `useSetupTradeState` to `CommonTradeUpdater` that is integrated into all trade widgets by default
2. Added `DisableNativeTokenSellingUpdater` that enables through params in `TradeWidget`
3. Moved other hooks to self updaters

  # To Test

1. Swap
- [ ] Set amounts from URL
- [ ] General swap flow
- [ ] ETH flow
2. Limit orders
- [ ] Set amounts from URL
- [ ] General limit orders flow
- [ ] Can't sell native token
3. Advanced orders
- [ ] Can't sell native token
- [ ] General TWAP flow

### CODE REVIEW

File: src/modules/trade/containers/TradeWidget/index.tsx [hunk 1]
```diff
@@ -125,6 +129,8 @@ export function TradeWidget(props: TradeWidgetProps) {
       <WrapNativeModal />
       <PriceImpactUpdater />
       <TradeFormValidationUpdater isExpertMode={isExpertMode} />
+      <CommonTradeUpdater />
+      {disableNativeSelling && <DisableNativeTokenSellingUpdater />}
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/modules/trade/containers/TradeWidget/index.tsx:133]: No, it should work as usual

**Comment by shoom3301** [comment]
@elena-zh I can't reproduce the case with quoting

**Comment by shoom3301** [comment]
@elena-zh thanks! Reproduced and fixed.
Also fixed https://github.com/cowprotocol/cowswap/issues/2718

File: src/legacy/state/swap/hooks.tsx [hunk 2]
```diff
@@ -252,9 +253,14 @@ export function useDerivedSwapInfo(): DerivedSwapInfo {
     [Field.OUTPUT]: { currencyId: outputCurrencyId },
     recipient,
   } = useSwapState()
+  const checkTokensWithSameSymbol = useAreThereTokensWithSameSymbol()
+
+  const inputCurrencyIsDoubled = checkTokensWithSameSymbol(inputCurrencyId)
+  const outputCurrencyIsDoubled = checkTokensWithSameSymbol(outputCurrencyId)
```
Comment (alfetopito) [inline src/legacy/state/swap/hooks.tsx:259]: `IsDuplicated` rather than `IsDoubled`?

**Comment by elena-zh** [comment]
Hey @shoom3301 , #2718 LGTM!
However, Swap form is not quoting trades at all now :(

File: src/legacy/state/swap/hooks.tsx (hunk 2 above)
Comment (shoom3301) [reply src/legacy/state/swap/hooks.tsx:259]: Thanks! Renamed

**Comment by shoom3301** [comment]
@elena-zh nice catch! Fixed

File: src/legacy/state/swap/hooks.tsx (hunk 2 above)
Comment (alfetopito) [reply src/legacy/state/swap/hooks.tsx:259]: Maybe you forgot to push?

**Pull request merged**